client/my-sites/checkout/composite-checkout/components/item-variation-picker/item-variation-dropdown.tsx

```diff
@@ -95,178 +14,38 @@ export const ItemVariationDropDown: FunctionComponent< ItemVariationPickerProps
-	const handleKeyDown: React.KeyboardEventHandler = useCallback(
-		( event ) => {
-			switch ( event.code ) {
-				case 'ArrowDown':
-					// prevent browser window from scrolling
-					event.preventDefault();
-					selectNextVariant();
-					break;
-				case 'ArrowUp':
-					// prevent browser window from scrolling
-					event.preventDefault();
-					selectPreviousVariant();
-					break;
-				case 'Enter':
-					event.preventDefault();
-					if (
-						highlightedVariantIndex !== null &&
-						highlightedVariantIndex !== selectedVariantIndex
-					) {
-						handleChange(
-							selectedItem.uuid,
-							variants[ highlightedVariantIndex ].productSlug,
-							variants[ highlightedVariantIndex ].productId
-						);
-					} else if ( highlightedVariantIndex === selectedVariantIndex ) {
-						toggleDropDown();
-					}
-					break;
-				case 'Space':
-					event.preventDefault();
-					toggleDropDown();
-					break;
-			}
-		},
-		[
-			handleChange,
-			highlightedVariantIndex,
-			selectedItem.uuid,
-			selectedVariantIndex,
-			selectNextVariant,
-			selectPreviousVariant,
-			toggleDropDown,
-			variants,
-		]
-	);
-
 	if ( variants.length < 2 ) {
 		return null;
 	}
 
-	return (
-		<Dropdown aria-expanded={ open } aria-haspopup="listbox" onKeyDown={ handleKeyDown }>
-			<CurrentOption
-				aria-label={ translate( 'Pick a product term' ) }
-				disabled={ isDisabled }
-				onClick={ () => setOpen( ! open ) }
-				open={ open }
-				role="button"
-			>
-				{ selectedVariantIndex !== null ? (
-					<ItemVariantPrice variant={ variants[ selectedVariantIndex ] } />
-				) : (
-					<span>{ translate( 'Pick a product term' ) }</span>
-				) }
-				<Gridicon icon={ open ? 'chevron-up' : 'chevron-down' } />
-			</CurrentOption>
-			{ open && (
-				<ItemVariantOptionList
-					variants={ variants }
-					highlightedVariantIndex={ highlightedVariantIndex }
-					selectedItem={ selectedItem }
-					handleChange={ handleChange }
-				/>
-			) }
-		</Dropdown>
-	);
-};
-
-function ItemVariantOptionList( {
-	variants,
-	highlightedVariantIndex,
-	selectedItem,
-	handleChange,
-}: {
-	variants: WPCOMProductVariant[];
-	highlightedVariantIndex: number | null;
-	selectedItem: ResponseCartProduct;
-	handleChange: ( uuid: string, productSlug: string, productId: number ) => void;
-} ) {
 	const compareTo = variants.find( ( variant ) => variant.productId === selectedItem.product_id );
-	return (
-		<OptionList role="listbox" tabIndex={ -1 }>
-			{ variants.map( ( variant, index ) => (
-				<ItemVariantOption
-					key={ variant.productSlug + variant.variantLabel }
-					isSelected={ index === highlightedVariantIndex }
-					onSelect={ () =>
-						handleChange( selectedItem.uuid, variant.productSlug, variant.productId )
-					}
-					compareTo={ compareTo }
-					variant={ variant }
-				/>
-			) ) }
-		</OptionList>
-	);
-}
+	const options = variants
+		.sort( ( { productId } ) => ( productId === selectedItem.product_id ? -1 : 1 ) )
```

Comment:
Controlling `CustomSelectControl`, i.e. passing the value as a prop (see [component README](https://github.com/Automattic/gutenberg/tree/trunk/packages/components/src/custom-select-control)) caused the selector to not update properly. I don't know why that what the case exactly.

The component keeps the current value in its own internal state, and calls `onChangeItemVariant` when it changes. To make sure the initially selected item is actually selected in the term selector, we make sure it's the first item passed as an option.

Reply:
🤔 Will this have an effect on the order of the variants displayed when the selector is open? What I mean is, currently if you have an annual plan, the selector will show the monthly variant first, then the annual, then the biennial; will this cause the monthly plan to be displayed second? If so, that's not necessarily a problem but it is worth considering the UX effects.

> Controlling CustomSelectControl, i.e. passing the value as a prop (see [component README](https://github.com/Automattic/gutenberg/tree/trunk/packages/components/src/custom-select-control)) caused the selector to not update properly. I don't know why that what the case exactly.

This seems like a pretty major bug for that component. Perhaps we shouldn't be relying on it if that bug can't be fixed.

Reply:
> Controlling CustomSelectControl, i.e. passing the value as a prop (see [component README](https://github.com/Automattic/gutenberg/tree/trunk/packages/components/src/custom-select-control)) caused the selector to not update properly. I don't know why that what the case exactly.

I'd like to come back to this. I had to put this PR on the back burner for a little while, and needed to refresh my memory. I've just tested a controlled version of `CustomSelectControl` and from a sighted user's perspective everything behaves as expected. However, the screen reader reads the outdated value after the term is changed. I've tested an implementation with a simple controlled `select` that leads to the same results. `CustomSelectControl` doesn't seem to be the issue here.

I suspect the screen reader (VoiceOver in this instance) being thrown off by the full line-item (and therefore the term selector) being re-rendered in the DOM.